### PR TITLE
Am 38

### PR DESCRIPTION
**Pull request by alphonse92**
Issue #38: resolved.

### CODE REVIEW

File: development/modules/users/templates/activity.html [hunk 1]
```diff
@@ -40,47 +40,47 @@
...
+            <img ng-src="{{curso.images._370x240}}" alt="">
           </a>
           <div class="description">
-            <a href="javascript:void(0)"
-               ng-click="showCourse($paginator_item)">
-              <h4>Curso Vital de Programación Neuro-Lingüistica</h4>
+            <a href="javascript:void(0)" ui-sref="courses.show({title:curso.name,course:curso,courseId:curso.id})">
+              <h4>{{curso.name}}</h4>
             </a>
             <h6>Marco Lino Alvarez Burgos</h6>
```
Comment (garusis) [inline development/modules/users/templates/activity.html]: Why is this hard coded? You should get the instructor.

File: development/modules/users/templates/activity.html [hunk 2]
```diff
@@ -40,47 +40,47 @@
...
+                <h4>{{$paginator_item.publication.title}}</h4>
+              </a>
+              <a ng-show="!$paginator_item.publication.active" href="javascript:void(0)">
+                <h4>Publicacion eliminada</h4>
               </a>
-              <h6>Marco Lino Alvarez Burgos | Publicado: 13 Feb 2017 Comentarios: 2</h6>
-              <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor
-                incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam...</p>
+
+              <h6 ng-show="$paginator_item.publication.active">Marco Lino Alvarez Burgos | Publicado: 13 Feb 2017 Comentarios: 2</h6>
```
Comment (garusis) [inline development/modules/users/templates/activity.html]: Hard coded again.

File: development/modules/students/servicesStudents.js [hunk 3]
```diff
@@ -66,6 +120,13 @@
         })
     }
 
+    CourseStudentRelation.prototype.getAll = function () {
```
Comment (garusis) [inline development/modules/students/servicesStudents.js]: Please, follow the standard in others relation builders. This method should be name **_get_** and give you the option to use a custom filter. https://github.com/garusis/marint_web/blob/4496c9c47f7912f99ea9b12dcb20ca77258ea081/development/modules/students/servicesStudents.js#L86

**Review by garusis — CHANGES_REQUESTED** [review]
Please, remove hard coded items and check the relation's implementation that exists. I'm fixing them right now, but take them in mind.

File: development/modules/students/servicesStudents.js [hunk 4]
```diff
@@ -36,9 +37,62 @@
...
+              active: false,
+              isPublicationLoaded: false
+            };
+            comment.getPublication = function () {
+              if (!comment.publication.isPublicationLoaded) {
+                var path = originsManager.getOrigin() +
+                  "/" + ROUTES.PUBLICATIONS.__BASE__ +
+                  "/" + comment.publication_id;
+                $http.get(path)
+                  .error(function (error) {
```
Comment (garusis) [inline development/modules/students/servicesStudents.js]: Error is not an ES Promises standard method. Please use then and catch at Promises.

File: development/modules/students/servicesStudents.js [hunk 5]
```diff
@@ -36,9 +37,62 @@
...
+    CommentStudentRelation.prototype.getComments = function () {
+      return $http.get(this.basePath)
+        .then(function (response) {
+          response.data.forEach(function (comment) {
+
+            comment.publication = {
+              active: false,
+              isPublicationLoaded: false
+            };
+            comment.getPublication = function () {
```
Comment (garusis) [inline development/modules/students/servicesStudents.js]: Why you're not using NewsService if that have all you need to that the same? Please, do not reinvent the wheel, be modular.

File: development/modules/users/templates/activity.html (hunk 1 above)
Comment (alphonse92) [reply development/modules/users/templates/activity.html]: I forget that

File: development/modules/students/servicesStudents.js (hunk 3 above)
Comment (alphonse92) [reply development/modules/students/servicesStudents.js]: resolved.

**Review by garusis — APPROVED** [review]

**Pull request merged**